Userland/Applications/PixelPaint/MainWidget.cpp

```diff
@@ -168,7 +169,9 @@ void MainWidget::initialize_menubar(GUI::Window& window)
                     return;
                 m_tab_widget->on_tab_close_click(*active_widget);
             }));
-    file_menu.add_action(GUI::CommonActions::make_quit_action([](auto&) {
+    file_menu.add_action(GUI::CommonActions::make_quit_action([this](auto&) {
+        if (!request_close())
```

Comment:
nit: I think it would be cleaner if we just call quit if we were requested to close, negative Boolean logic isn't needed to express this. 
```suggestion
        if (request_close())
            GUI::Application::the()->quit();
```